src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -262,69 +316,120 @@ private async Task<string> DownloadAsync(Uri zipUri)
 
         private void UnpackPackage(string filePath, string scriptPath)
         {
-            if (_environment.IsMountEnabled() &&
-                // Only attempt to use FUSE on Linux
-                RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            var packageType = GetPackageType(filePath);
+
+            if (packageType == CodePackageType.Squashfs)
             {
-                using (_metricsLogger.LatencyEvent(MetricEventNames.LinuxContainerSpecializationFuseMount))
+                // default to mount for squashfs images
+                if (_environment.IsMountDisabled())
                 {
-                    if (FileIsAny(".squashfs", ".sfs", ".sqsh", ".img", ".fs"))
-                    {
-                        MountFsImage(filePath, scriptPath);
-                    }
-                    else if (FileIsAny(".zip"))
-                    {
-                        MountZipFile(filePath, scriptPath);
-                    }
-                    else
-                    {
-                        throw new InvalidOperationException($"Can't find Filesystem to match {filePath}");
-                    }
+                    UnsquashImage(filePath, scriptPath);
+                }
+                else
+                {
+                    MountSquashfsImage(filePath, scriptPath);
                 }
             }
-            else
+            else if (packageType == CodePackageType.Zip)
             {
-                using (_metricsLogger.LatencyEvent(MetricEventNames.LinuxContainerSpecializationZipExtract))
+                // default to unzip for zip packages
+                if (_environment.IsMountEnabled())
                 {
-                    _logger.LogInformation($"Extracting files to '{scriptPath}'");
-                    ZipFile.ExtractToDirectory(filePath, scriptPath, overwriteFiles: true);
-                    _logger.LogInformation($"Zip extraction complete");
+                    MountZipFile(filePath, scriptPath);
                 }
+                else
+                {
+                    UnzipPackage(filePath, scriptPath);
+                }
+            }
+        }
+
+        private CodePackageType GetPackageType(string filePath)
+        {
+            // Try checking the file extension
+            if (FileIsAny(".squashfs", ".sfs", ".sqsh", ".img", ".fs"))
+            {
+                return CodePackageType.Squashfs;
+            }
+            else if (FileIsAny(".zip"))
+            {
+                return CodePackageType.Zip;
+            }
+
+            // No file extension match found. Try checking magic number using `file` command.
+            (var output, _, _) = RunBashCommand($"file -b {filePath}", MetricEventNames.LinuxContainerSpecializationFileCommand);
```

Comment:
Can we default the package type detection by `file -b {filePath}` instead of using file extension?
In Linux Consumption, the SCM_RUN_FROM_PACKAGE path will always have a **.zip** file extension.

Reply:
It would be confusing if the file ends in the wrong extension. It'll appear corrupted when you try to extract it if you don't know that. Why do they always end in zip?

Reply:
As we discussed in this morning, we will always replace the .zip package in storage account instead of uploading to a new one.